### PR TITLE
Fix some clippy warnings

### PR DESCRIPTION
**Pull request by mikhailOK**
clippy works again

### CODE REVIEW

File: core/primitives/src/crypto/signature.rs [hunk 1]
```diff
@@ -47,20 +47,6 @@ impl ToBytes for SecretKey {
     }
 }
 
-/// Allows using PublicKey in std containers and macros.
```
Comment (bowenwang1996) [inline core/primitives/src/crypto/signature.rs]: Why do we remove this? Is it because this is not used?

Comment (mikhailOK) [reply core/primitives/src/crypto/signature.rs]: there is a clippy error: derived equality but custom hash

Comment (bowenwang1996) [reply core/primitives/src/crypto/signature.rs]: Wait then we should implement `PartialEq` instead of removing this

Comment (mikhailOK) [reply core/primitives/src/crypto/signature.rs]: it's also unused

**Pull request merged**